Ensure exported identity cards include certificates and private

jt-nti opened the pull request:
Ensure exported identity cards include certificates and private
keys when they are available.

Also includes some refactoring to simplify how cards are activated
and minimise the time that credentials are kept in the browser.

Completes #1427

Signed-off-by: James Taylor <jamest@uk.ibm.com>

caroline-church commented on packages/composer-playground/src/app/services/identity-card.service.ts:
```diff
@@ -277,24 +292,40 @@ export class IdentityCardService {
+                hasCredentials = true;
+
+                // don't want to keep credentials around after the card has been activated
+                card.setCredentials({});
```
We probably shouldn't keep the secret either once they have enrolled as its not needed, should set it to empty string

jt-nti replied:
You potentially need the secret if the card hasn't been used anywhere so I'm not quite sure where this should happen- possibly in the admin service after connecting?
Removing the secret would be nice but in a real system it should only be good for a single use, and there's no harm it remaining in the cards after use.

jt-nti commented:
I've added some info to the doc regarding the implications of this PR, and renamed public key to certificate. I don't think the current doc would be sufficient even with the correct file names however that is not related to this PR- I'll add something to the existing ID card doc issue #1983
